specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-11-01-preview/AssistedQueries.json

```diff
@@ -0,0 +1,612 @@
+          },
+          {
+            "$ref": "#/parameters/EditedQueryId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Creates or updates an edited query.": {
+            "$ref": "./examples/assistedQueries/CreateEditedQuery.json"
+          }
+        },
+        "tags": [
+          "EditedQueries"
+        ],
+        "description": "Creates or updates an edited query.",
+        "operationId": "EditedQueries_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/QueryId"
+          },
+          {
+            "$ref": "#/parameters/EditedQueryId"
+          },
+          {
+            "$ref": "#/parameters/EditedQuery"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/EditedQuery"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/EditedQuery"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "QueryList": {
+      "description": "List all the queries.",
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of queries.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of queries.",
+          "items": {
+            "$ref": "#/definitions/Query"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "Query": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
```

Comment:
>ResourceWithEtag [](http://example.com/codeflow?start=63&length=16)

An extension resource cannot be tracked and cannot have location 

[https://armwiki.azurewebsites.net/rp_onboarding/extension_resources.html](https://armwiki.azurewebsites.net/rp_onboarding/extension_resources.html)

Reply:
I followed the way the other existing APIs in the same directory defined their schema.

Reply:
Which other existing apis , please read through the documentation and model this correctly. The other apis must have been explicitly designed as tracked resources. Your resource is an extension resource which cannot be tracked and cannot have a location or tags